rfc/20250207-sbom.md

```diff
@@ -0,0 +1,184 @@
+    artifacts: binary
+```
+
+## 2. Enabling publication of SBOM artifacts in the OpenTofu Registry
+
+### Protocol changes
+
+To support the discovery of SBOM artifacts, we propose changing the ["Find a Provider Package"](https://opentofu.org/docs/internals/provider-registry-protocol/#find-a-provider-package) endpoint located at `:namespace/:type/:version/download/:os/:arch` to include the `attachments` field:
+
+```json
+{
+  "protocols": ["4.0", "5.1"],
+  "os": "linux",
+  "arch": "amd64",
+  "filename": "terraform-provider-random_2.0.0_linux_amd64.zip",
+  "download_url": "https://releases.example.com/terraform-provider-random/2.0.0/terraform-provider-random_2.0.0_linux_amd64.zip",
+  "shasums_url": "https://releases.example.com/terraform-provider-random/2.0.0/terraform-provider-random_2.0.0_SHA256SUMS",
+  "shasums_signature_url": "https://releases.example.com/terraform-provider-random/2.0.0/terraform-provider-random_2.0.0_SHA256SUMS.sig",
+  "shasum": "5f9c7aa76b7c34d722fc9123208e26b22d60440cb47150dd04733b9b94f4541a",
+  "signing_keys": {
+    "gpg_public_keys": [
+      {
+        "key_id": "51852D87348FFC4C",
+        "ascii_armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\nVersion: GnuPG v1\n\nmQENBFMO...",
+        "trust_signature": "",
+        "source": "ExampleCorp",
+        "source_url": "https://www.examplecorp.com/security.html"
+      }
+    ]
+  },
+  "attachments": [
+    {
+      "media_type": "application/spdx+json",
+      "filename": "sbom.spdx.json",
+      "download_url": "https://releases.example.com/terraform-provider-random/2.0.0/sbom.spdx.json"
+    }
+  ]
+}
+```
+
+The OpenTofu Registry should perform a *narrowly tailored* scan for SBOM and other attestation artifacts and include them in the `attachments` section according to the following rules:
+
+- `*.spdx.json` and `*.spdx.sbom.json` (SPDX) will be identified as [`application/spdx+json`](https://www.iana.org/assignments/media-types/application/spdx+json)
+
+However, we are currently excluding the following:
+
+- `*.spdx.xml`, `*.spdx.yml`, and `*.spdx.yaml` (SPDX) will *not* be supported until [there is an approved MIME type for them](https://github.com/spdx/spdx-spec/issues/577#issuecomment-960295523)
+- `bom.xml` and `bom.json` (CycloneDX) will *not* be supported [until there is an approved MIME type for it](https://github.com/CycloneDX/specification/issues/210)
+
+These rules may be amended later if there is sufficient proof that a correct and unique MIME type has been established by the format vendor.
+
+Since the response is a per-platform response, the OpenTofu registry should follow the naming rules for the provider binaries to determine if an attestation/SBOM artifact is related to a single platform or to all platforms. In the registry response both kinds of responses should be included. For example:
+
+- `terraform-provider-aws_5.86.0_linux_arm.spdx.json` relates to the `linux/arm` platform
+- `slsa.in-toto.jsonl` relates to all platforms

```

Comment:
The rest of this proposal is focused only on the SPDX-format SBOM and doesn't include in-toto attestations, so this appearing here is perhaps a little confusing, but I understand that we don't actually _have_ any non-platform-specific files right now.

For the sake of this RFC in particular, could we just state that any file without a recognizable platform identifier is ignored? A future RFC can define other naming schemes that don't include platform identifiers if needed.